dgraphtest/acl.go

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright 2023 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dgraphtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/dgraph-io/dgraph/testutil"
+	"github.com/pkg/errors"
+)
+
+var (
+	errGroupCountMoreThanOne = errors.New("group count is other than 1")
+)
+
+type AclGroupRules struct {
+	Predicate  string `json:"predicate"`
+	Permission int32  `json:"permission"`
+}
+type AclGroup struct {
+	Name  string          `json:"name"`
+	Rules []AclGroupRules `json:"rules"`
+}
+
+func (hc *HTTPClient) GetCurrentUser() (string, error) {
+	const query = `
+	query {
+		getCurrentUser {
+			name
+		}
+	}`
+	params := GraphQLParams{
+		Query: query,
+	}
+	resp, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return string(resp), err
+	}
+	var userResp struct {
+		GetCurrentUser struct {
+			Name string
+		}
+	}
+	if err := json.Unmarshal(resp, &userResp); err != nil {
+		return "", errors.Wrapf(err, "error unmarshalling getCurrentUser response")
+	}
+	return userResp.GetCurrentUser.Name, nil
+}
+
+func (hc *HTTPClient) DeleteUser(username string) error {
+	delUser := `mutation deleteUser($name: String!) {
+		deleteUser(filter: {name: {eq: $name}}) {
+			msg
+			numUids
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: delUser,
+		Variables: map[string]interface{}{
+			"name": username,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	return err
+}
+
+func (hc *HTTPClient) CreateUser(username, password string) (string, error) {
+	addUser := `mutation addUser($name: String!, $pass: String!) {
+		addUser(input: [{name: $name, password: $pass}]) {
+			user {
+				name
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addUser,
+		Variables: map[string]interface{}{
+			"name": username,
+			"pass": password,
+		},
+	}
+	resp, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return "", err
+	}
+	type Response struct {
+		AddUser struct {
+			User []struct {
+				Name string
+			}
+		}
+	}
+
+	var r Response
+	if err := json.Unmarshal(resp, &r); err != nil {
+		return "", err
+	}
+	return r.AddUser.User[0].Name, nil
```

Comment:
would it be a good idea to check for length of ` r.AddUser.User` etc. (in other places) before indexing into 0 elemnt?